control/subscription_group.go

```diff
@@ -386,72 +389,53 @@ func (s *subscriptionGroup) process(id string) (serrs []serror.SnapError) {
 
 func (s *subscriptionGroup) subscribePlugins(id string,
 	plugins []core.SubscribedPlugin) (serrs []serror.SnapError) {
-	for _, sub := range plugins {
+	plgs := make([]*loadedPlugin, len(plugins))
+	// First range through plugins to verify if all required plugins
+	// are available
+	for i, sub := range plugins {
+		plg, err := s.pluginManager.get(key(sub))
+		if err != nil {
+			fmt.Println(err)
+			serrs = append(serrs, serror.New(err))
+			return
+		}
+		plgs[i] = plg
+	}
+
+	// If all plugins are available, subscribe to pools and start
+	// plugins as needed
+	for _, plg := range plgs {
 		controlLogger.WithFields(log.Fields{
-			"name":    sub.Name(),
-			"type":    sub.TypeName(),
-			"version": sub.Version(),
+			"name":    plg.Name(),
+			"type":    plg.TypeName(),
+			"version": plg.Version(),
 			"_block":  "subscriptionGroup.subscribePlugins",
 		}).Debug("plugin subscription")
-		if sub.Version() < 1 {
```

Comment:
No functionality lost. Currently, the workflow for a plugin with no version specified (-1) is to get the plugin, then get or create the pool associated with that plugin. If a version was specified, the code would only get the plugin if the pool was eligible to be grown. The change is here to get all the plugins first, then if we all the plugins are available, subscribe to the pools, etc.